### PR TITLE
[ADD] ci: enable tests using GitHub actions t#54296

### PR DESCRIPTION
**Pull request by luisg123v**
The following tests are added:
- pre-commit-vauxoo
- No dependency files
- Odoo

### CODE REVIEW

**Comment by luisg123v** [comment]
@moylop260 

Could you review, please?

I know it's red, but that's why I wanted to enable it. Next step is to fix those reds.

Regards,

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    name: ${{ matrix.name }}
+    env:
+      PRECOMMIT_HOOKS_TYPE: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [mandatory, optional]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Upgrade pre-commit-vauxoo to te latest version
```
Comment (moylop260) [inline .github/workflows/pre-commit.yml]:
```suggestion
      - name: Upgrade pre-commit-vauxoo to the latest version
```

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        run: pip3 install --upgrade pre-commit-vauxoo
+      - name: Run pre-commit-vauxoo
+        run: pre-commit-vauxoo
```
Comment (moylop260) [inline .github/workflows/pre-commit.yml]: How are you set optional or mandatory to the command?

**Review by moylop260 — COMMENTED** [review]
The OPTIONAL pipeline is showing green status

Shouldn't be yellow one?

File: .github/workflows/pre-commit.yml (hunk 2 above)
Comment (luisg123v) [reply .github/workflows/pre-commit.yml]: Setting variable from job name:

```
env:
  PRECOMMIT_HOOKS_TYPE: ${{ matrix.name }}
```

**Comment by moylop260** [comment]
Could you merge the build pre-commit-optional and pre-commit-mandatory in order to review all the logs only in one build even if it is green, please?

Since github doesn't have yellow status or allow_failure as gitlab so the optional will be green all the time but we have 2 pre-commit builds so the developer will need to check the log of optional but they could get confused about what pre-commit build check

IMHO it is better if you have the full logs only in one build for this case in order to check only one output instead of 2

**Comment by luisg123v** [comment]
@moylop260 I agree, I wasn't aware GitHub doesn't support yellow when I implemented this

**Comment by moylop260** [comment]
excellent!

Also, you even don't need the container to run the tests

I think it could be even faster running over the original GH ubuntu but with cache

I will make a PR to your PR...

**Comment by moylop260** [comment]
Please, check the following changes:
 - https://github.com/Vauxoo/addons-vauxoo/pull/1564

It is running the pre-commit build spending 23s

**Comment by moylop260** [comment]
Also, could you add a error if the dev adds `oca_dependencies.txt` file in the CI, please?

**Comment by moylop260** [comment]
👍 
(After removing the configuration files)

**Pull request merged**